### ISSUE TITLE
cloudflare_worker_route.script_name prompts for update even with no changes

### ISSUE THREAD

**Issue by ivanschwarz** [terraform-providers/terraform-provider-cloudflare#621]
### Terraform Version
Terraform v0.12.23

### Provider Version
terraform-provider-cloudflare_v2.4.0_x4

### Affected Resource(s)
cloudflare_worker_route

### Terraform Configuration Files
```hcl
resource "cloudflare_worker_route" "my_route" {
  zone_id     = "..."
  pattern     = "..."
  script_name = "..."
}
```

### Expected Behavior
No changes

### Actual Behavior
```hcl
  ~ resource "cloudflare_worker_route" "my_route" {
        id          = "..."
        pattern     = "..."
      + script_name = "..."
        zone_id     = "..."
    }
```

### Steps to Reproduce
1. `terraform apply` (to create the route)
2. `terraform apply` (to see that a plan to update `script_name` is generated)
3. `terraform apply` (to see that a plan to update `script_name` is generated again)

### Important Factoids
Running `terraform apply` on the same configuration multiple times should be an idempotent operation, but in this case it always produces a plan to update the `script_name` even if previous plan succeeded and I can see a correct script assigned to the worker route in the Cloudflare console.

I've checked the state and there I can see the `script_name` correctly stored.

**Comment by jacobbednarz** [comment]
Hi @ivanschwarz, I'm afraid I've tried to reproduce this with the following and I've been unable to do so.

```js
// script.js
addEventListener("fetch", event => {
  event.respondWith(handleRequest(event.request));
});
/**
 * Respond with hello worker text
 * @param {Request} request
 */
async function handleRequest(request) {
  return new Response("Hello worker!", {
    headers: { "content-type": "text/plain" }
  });
}

```

```hcl
provider "cloudflare" {
  version    = "~> 2.0"
  email      = var.cloudflare_email
  api_key    = var.cloudflare_api_key
  account_id = var.cloudflare_account_id
}

variable "cloudflare_email" {}
variable "cloudflare_api_key" {}
variable "cloudflare_account_id" {}
variable "cloudflare_zone_id" {}
variable "cloudflare_zone_name" {}

resource "cloudflare_worker_script" "my_script" {
  name    = "example"
  content = file("script.js")
}

resource "cloudflare_worker_route" "my_route" {
  zone_id     = var.cloudflare_zone_id
  pattern     = "${var.cloudflare_zone_name}/*"
  script_name = cloudflare_worker_script.my_script.name
}
```

<details><summary>Initial output</summary>

```
$ terraform init && \
  terraform apply -var 'cloudflare_email=REDACTED' -var 'cloudflare_api_key=REDACTED' -var 'cloudflare_zone_id=REDACTED' -var 'cloudflare_zone_name=example.com' -var 'cloudflare_account_id=REDACTED'

Initializing the backend...

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "cloudflare" (terraform-providers/cloudflare) 2.4.1...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # cloudflare_worker_route.my_route will be created
  + resource "cloudflare_worker_route" "my_route" {
      + id          = (known after apply)
      + pattern     = "example.com/*"
      + script_name = "example"
      + zone_id     = "REDACTED"
    }

  # cloudflare_worker_script.my_script will be created
  + resource "cloudflare_worker_script" "my_script" {
      + content = <<~EOT
            addEventListener("fetch", event => {
              event.respondWith(handleRequest(event.request));
            });
            /**
             * Respond with hello worker text
             * @param {Request} request
             */
            async function handleRequest(request) {
              return new Response("Hello worker!", {
                headers: { "content-type": "text/plain" }
              });
            }
        EOT
      + id      = (known after apply)
      + name    = "example"
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

cloudflare_worker_script.my_script: Creating...
cloudflare_worker_script.my_script: Creation complete after 2s [id=example]
cloudflare_worker_route.my_route: Creating...
cloudflare_worker_route.my_route: Creation complete after 1s [id=243c05312c8f4c719afd6da1672b7393]
```

</details>

<details><summary>Reapply output</summary>

```
$ terraform init && \
  terraform apply -var 'cloudflare_email=REDACTED' -var 'cloudflare_api_key=REDACTED' -var 'cloudflare_zone_id=REDACTED' -var 'cloudflare_zone_name=example.com' -var 'cloudflare_account_id=REDACTED'

Initializing the backend...

Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
cloudflare_worker_script.my_script: Refreshing state... [id=example]
cloudflare_worker_route.my_route: Refreshing state... [id=243c05312c8f4c719afd6da1672b7393]

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

</details>

Are you able to please try my reproduction case and let me know if you still have an issue? If we can't reproduce with this setup, I'd be looking at a local issue to your installation that is causing the problem.

**Comment by ivanschwarz** [comment]
@jacobbednarz Thank you for taking time to try to reproduce my issue. I've used your configuration and it indeed doesn't suffer from the issue I described in the OP. I've compared your configuration to our configuration to single out the culprit.

In our configuration we actually don't use `account_id` in the provider definition:
```hcl
provider "cloudflare" {
  version = "2.4.0"
  email   = "..."
  api_key = "..."
}
```

We didn't experience any problems while omitting `account_id` with exception of the issue described in the OP. If I add `account_id`, the issue goes away. 

I assume the correct way is to actually include `account_id`, so the problem is on our side. Should it be maybe required then? Anyway feel free to close this issue and again, thank you for your time.

**Comment by jacobbednarz** [comment]
I suspect this is the underlying issue; https://github.com/terraform-providers/terraform-provider-cloudflare/blob/c3b855c2402d652470b04243c0d473de98ba8612/vendor/github.com/cloudflare/cloudflare-go/workers.go#L603

Various parts of Workers used to be account level (and some enterprise) only bits of functionality. That has since been removed but I suspect there are still parts of the underlying library that aren't yet taking this into account due to not being able to deprecate it.

We have added some documentation for the [Workers KV resource](https://www.terraform.io/docs/providers/cloudflare/r/workers_kv.html) and I'd happily review a PR adding that for the other resources impacted here too if you want to take a pass at it.
